File: client/Platform/src/app/features/internship-management/core/models/internship-settings.ts
import { InternshipSettingsResponse } from "../interfaces/internship-settings";

export class InternshipSettings {
    public id: string;
    public status: string;
    public maximumInternsToEnroll: number;
    public durationInMonths: number;
    public scheduledToStartOn: Date;
    public scheduledToEndOn: Date;
    public center: string;

    constructor(response: InternshipSettingsResponse)
    {
        this.id = response.id;
        this.durationInMonths = response.durationInMonths;
        this.maximumInternsToEnroll = response.maximumInternsToEnroll;
        this.status = "";
        this.scheduledToStartOn = response.scheduledToStartOn;
        this.scheduledToEndOn = response.scheduledToEndOn;
        this.center = response.center;
    }
}

File: backend/ProgrammingInternshipPlatform/ProgrammingInternshipPlatform.Application/InternshipManagement/GetInternshipSettings/Responses/InternshipSettings.cs
using ProgrammingInternshipPlatform.Domain.InternshipManagement.Internships;

namespace ProgrammingInternshipPlatform.Application.InternshipManagement.GetInternshipSettings.Responses;

public class InternshipSettings
{
    public Guid Id { get; set; }
    public InternshipStatus Status { get; }
    public int MaximumInternsToEnroll { get; }
    public int DurationInMonths { get; }
    public DateTime ScheduledToStartOn { get; }
    public DateTime ScheduledToEndOn { get; }
    public string Center { get; }

    public InternshipSettings(Guid id, InternshipStatus status, int maximumInternsToEnroll, int durationInMonths,
        DateTime scheduledToStartOn, DateTime scheduledToEndOn, string center)
    {
        Id = id;
        Status = status;
        MaximumInternsToEnroll = maximumInternsToEnroll;
        DurationInMonths = durationInMonths;
        ScheduledToStartOn = scheduledToStartOn;
        ScheduledToEndOn = scheduledToEndOn;
        Center = center;
    }
}